docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+
+With that in mind, we create a "naive" version of our text classifier tool.
+That is, a tool without any UI to ensure critical functions are running as intended.
+To achieve this, we also mock data and relevant calls to represent what will come from the UI SDK to the Python backend and what the backend will send to the UI SDK via `save_config()` (link to docs here). 
+Next, we build our front end using the UI SDK using its provided `dev-harness`.
+Once the back and front ends are complete, we then bundle and deploy using the CLI to do end-to-end testing.
+Finally, we package our dependencies and source for deployment using `create-yxi`.
+
+The tool will have 4 "modes": `DATA`, `PREVIEW`, `TRAIN`, and `PREDICT`. 
+
+### 1.2.1. Data
+
+In `DATA` mode, our tool will pull down and prep whatever data the User provides via input fields we'll define.
+It will then save the data as a TensorFlow dataset, ready for the next mode.
+
+### 1.2.2. Preview
+
+In `PREVIEW` mode, our tool will then use the data provided and any available model information to show a "snapshot" of our data and how we will feed it to the model.
+This snapshot will include dynamically populated text classifier information like token translations for our classifier when available.
+
+### 1.2.3. TRAIN
+
+This mode will take the prepped data and call the model's training function.
+Additionally, the User may use the supplied fields to adjust model parameters, train using an existing model, or create an entirely new one.
+Finally, the tool will return a training results report to the UI and export a "production" version of the model.
+
+### 1.2.4. Predict
+
+In `PREDICT` mode, the tool will take text input via the input anchor, then run predictions on them and output those results to its output anchor.
+
+
+## Create a Workspace
+
+Run the following, responding as prompted.
+
+```powershell
+$ mkdir tensorflow-ui-examples
+$ cd ./tensorflow-ui-examples/
+tensorflow-ui-examples $ ayx_plugin_cli sdk-workspace-init
+Package Name: PythonSdkTensorflowExamples
+Tool Category [Python SDK Examples]: Tensorflow Python SDK Examples
+Description []: Tools using Tensorflow.
+Author []: Shawn Owen
+Company []: Alteryx
+Backend Language (python): python
+[Workspace initialization] started
+[Workspace initialization] .  Create configuration directory
+// output truncated //
+[Generating config files] started
+[Generating config files] .  generate_config_files:generate_config_xml
+[Generating config files] Generating top level config XML file...
+[Generating config files] finished
+```
+Once the above is complete, we recommend you create your Python environment with `venv` or other supported virtual environment modules.
+This venv will allow you to develop on your machine while deploying with "prod" requirements. We'll also briefly cover how to do that later in the guide.
```

Comment:
TW review:

Once the above is complete, we recommend you create your Python environment with `venv` or other supported virtual environment modules. This venv allows you to develop on your machine while deploying with "prod" requirements. We also briefly cover how to do that later in this guide.